modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -0,0 +1,502 @@
+
+        if (SUPER_TENANT.equals(tenant)) {
+            expectedResponse = readResource("list-meta-outbound-connectors-response.json");
+            jsonWriter = new ObjectMapper(new JsonFactory());
+            metaOutboundConnectorListResponse = Arrays.asList(jsonWriter.readValue(expectedResponse,
+                    MetaOutboundConnectorListItem[].class));
+        } else {
+            expectedResponse = readResource("list-meta-outbound-connectors-tenant-response.json");
+            jsonWriter = new ObjectMapper(new JsonFactory());
+            metaOutboundConnectorListResponse = Arrays.asList(jsonWriter.readValue(expectedResponse,
+                    MetaOutboundConnectorListItem[].class));
+        }
+
+        expectedResponse = readResource("get-meta-outbound-connector-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        metaOutboundConnectorResponse = jsonWriter.readValue(expectedResponse, MetaOutboundConnector.class);
+
+        expectedResponse = readResource("get-claims-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        claimsResponse = jsonWriter.readValue(expectedResponse, Claims.class);
+
+        expectedResponse = readResource("get-roles-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        rolesResponse = jsonWriter.readValue(expectedResponse, Roles.class);
+
+        expectedResponse = readResource("get-jit-response.json");
+        jsonWriter = new ObjectMapper(new JsonFactory());
+        justInTimeProvisioningResponse = jsonWriter.readValue(expectedResponse, JustInTimeProvisioning.class);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testConclude() {
+
+        super.conclude();
+    }
+
+    @BeforeMethod(alwaysRun = true)
+    public void testInit() {
+
+        RestAssured.basePath = basePath;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void testFinish() {
+
+        RestAssured.basePath = StringUtils.EMPTY;
+    }
+
+    @DataProvider(name = "restAPIUserConfigProvider")
+    public static Object[][] restAPIUserConfigProvider() {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Test
+    public void testListMetaFederatedAuthenticators() throws IOException {
+
+        Response response = getResponseOfGet(IDP_API_BASE_PATH + PATH_SEPARATOR + META_FEDERATED_AUTHENTICATORS_PATH);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK);
+
+        ObjectMapper jsonWriter = new ObjectMapper(new JsonFactory());
+        List<MetaFederatedAuthenticatorListItem> responseFound =
+                Arrays.asList(jsonWriter.readValue(response.asString(), MetaFederatedAuthenticatorListItem[].class));
+        Assert.assertEquals(responseFound, metaFederatedAuthenticatorListResponse,
```

Comment:
Directly comparing lists will cause problems. Using JSON paths you have to compare the items.

Reply:
By doing so, you can get rid of the model objects.